cli/deployments/eirini.go

```diff
@@ -0,0 +1,285 @@
+		return err
+	}
+	if err = k.patchServiceAccountWithSecretAccess(c, "eirini"); err != nil {
+		return err
+	}
+
+	if err := c.WaitForPodBySelectorRunning("eirini-core", "name=eirini-api", 300); err != nil {
+		return errors.Wrap(err, "failed waiting Eirini api deployment to come up")
+	}
+	if err := c.WaitForPodBySelectorRunning("eirini-core", "name=eirini-metrics", 300); err != nil {
+		return errors.Wrap(err, "failed waiting Eirini metrics deployment to come up")
+	}
+
+	emoji.Println(":heavy_check_mark: Eirini deployed")
+
+	return nil
+}
+
+func (k Eirini) GetVersion() string {
+	return eiriniVersion
+}
+
+func (k Eirini) Deploy(c kubernetes.Cluster, options kubernetes.InstallationOptions) error {
+
+	_, err := c.Kubectl.CoreV1().Namespaces().Get(
+		context.Background(),
+		eiriniDeploymentID,
+		metav1.GetOptions{},
+	)
+	if err == nil {
+		return errors.New("Namespace " + eiriniDeploymentID + " present already")
+	}
+
+	emoji.Println(":ship:Deploying Eirini")
+	err = k.apply(c, options)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (k Eirini) Upgrade(c kubernetes.Cluster, options kubernetes.InstallationOptions) error {
+	_, err := c.Kubectl.CoreV1().Namespaces().Get(
+		context.Background(),
+		eiriniDeploymentID,
+		metav1.GetOptions{},
+	)
+	if err != nil {
+		return errors.New("Namespace " + eiriniDeploymentID + " not present")
+	}
+
+	emoji.Println(":ship:Upgrade Eirini")
+	return k.apply(c, options)
+}
+
+func (k Eirini) createClusterRegistryCredsSecret(c kubernetes.Cluster, http bool) error {
+	var protocol, secretName string
+	if http {
+		protocol = "http"
+		secretName = "cluster-registry-creds"
+	} else {
+		protocol = "https"
+		secretName = "cluster-registry-creds-http"
+	}
+	_, err := c.Kubectl.CoreV1().Secrets("eirini-workloads").Create(context.Background(),
+		&corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: secretName,
+			},
+			StringData: map[string]string{
+				".dockerconfigjson": `{"auths":{"` + protocol + `://127.0.0.1:30501":{"auth": "YWRtaW46cGFzc3dvcmQ=", "username":"admin","password":"password"}}}`,
+			},
+			Type: "kubernetes.io/dockerconfigjson",
+		}, metav1.CreateOptions{})
+
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (k Eirini) createGitCredsSecret(c kubernetes.Cluster, domain string) error {
+	_, err := c.Kubectl.CoreV1().Secrets("eirini-workloads").Create(context.Background(),
+		&corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "git-creds",
+				Annotations: map[string]string{
+					"kpack.io/git": fmt.Sprintf("http://%s.%s", giteaDeploymentID, domain),
+				},
+			},
+			StringData: map[string]string{
+				"username": "dev",
+				"password": "changeme",
+			},
+			Type: "kubernetes.io/basic-auth",
+		}, metav1.CreateOptions{})
+
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (k Eirini) patchServiceAccountWithSecretAccess(c kubernetes.Cluster, name string) error {
+	patchContents := `
+{
+  "secrets": [
```

Comment:
Nit: Inconsistent indent with the imagePullSecrets.

Reply:
They look the same in my vim (but one is spaces the other is tab). Fixing.

Reply:
Ah. Yes tab vs spaces is seen easily in various diff displays.